src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -0,0 +1,395 @@
+                {
+                    UInt32 offset = 0;
+                    Int32 bytesRead = 0;
+
+                    while ((bytesRead = reader.Read(buffer, 0, BUFFERSIZE)) > 0)
+                    {
+                        if (bytesRead == BUFFERSIZE)
+                        {
+                            // We are reusing the same buffer so if we save the output to a variable, the variable
+                            // will just contain multiple references to the same buffer memory space (containing only the
+                            // last bytes of the file read).  Cloning the buffer allows us to pass the values on without
+                            // overwriting previous values.
+                            byte[] bufferClone = (byte[])buffer.Clone();
+                            ConvertToHexidecimal(bufferClone, path, offset);
+                        }
+                        else
+                        {
+                            // Handle the case of a partial (and probably last) buffer.  Copies the bytes read into a new,
+                            // shorter array so we do not have the extra bytes from the previous pass through at the end.
+                            byte[] remainingBytes = new byte[bytesRead];
+                            Array.Copy(buffer, 0, remainingBytes, 0, bytesRead);
+                            ConvertToHexidecimal(remainingBytes, path, offset);
+                        }
+                        // Update offset value.
+                        offset += (UInt32)bytesRead;
+                    }
+                }
+            }
+            catch (IOException ioException)
+            {
+                // IOException takes care of FileNotFoundException, DirectoryNotFoundException, and PathTooLongException
+                WriteError(new ErrorRecord(ioException, "FormatHexIOError", ErrorCategory.WriteError, path));
+            }
+            catch (ArgumentException argException)
+            {
+                WriteError(new ErrorRecord(argException, "FormatHexArgumentError", ErrorCategory.WriteError, path));
+            }
+            catch (NotSupportedException notSupportedException)
+            {
+                WriteError(new ErrorRecord(notSupportedException, "FormatHexPathRefersToANonFileDevice", ErrorCategory.InvalidArgument, path));
+            }
+            catch (SecurityException securityException)
+            {
+                WriteError(new ErrorRecord(securityException, "FormatHexUnauthorizedAccessError", ErrorCategory.PermissionDenied, path));
+            }
+        }
+
+        #endregion
+
+        #region InputObjects
+
+        /// <summary>
+        /// Returns a new instance of the Encoding Class based on cmdlet Encoding parameter.
+        /// </summary>
+        /// <param name="encoding"></param>
+        /// <returns></returns>
+        private Encoding GetEncoding(string encoding)
```

Comment:
Why is the private member not camel cased ? Unless Pascal casing is the recommended pattern for C# cmdlets, we should use camel casing